File: Launcher/Downloader/src/main/java/info/smart_tools/smartactors/downloader/Feature.java
package info.smart_tools.smartactors.downloader;

import java.util.List;

public class Feature {
    private String featureName;
    private List<String> afterFeatures;
    private Repository repository;
    private List<String> plugins;


    public String getFeatureName() {
        return featureName;
    }

    public void setFeatureName(String featureName) {
        this.featureName = featureName;
    }

    public List<String> getAfterFeatures() {
        return afterFeatures;
    }

    public void setAfterFeatures(List<String> afterFeatures) {
        this.afterFeatures = afterFeatures;
    }

    public Repository getRepository() {
        return repository;
    }

    public void setRepository(Repository repository) {
        this.repository = repository;
    }

    public List<String> getPlugins() {
        return plugins;
    }

    public void setPlugins(List<String> plugins) {
        this.plugins = plugins;
    }
//
//    public String getName() {
//        return this.featureName.split(":")[1];
//    }
//
//    public String getGroup() {
//        return this.featureName.split(":")[0];
//    }
//
//    public String getVersion() {
//        return this.featureName.split(":")[2];
//    }
}

File: DevTools/SmartActorsIdeaPlugin/src/main/kotlin/info/smart_tools/smartactors/devtools/common/feature/Feature.kt
package info.smart_tools.smartactors.devtools.common.feature

data class Repository(val id: String, val url: String)

data class FeatureName(
    val groupId: String,
    val name: String,
    val version: String?
) {
    override fun toString(): String {
        return "$groupId:$name${if (version != null) ":$version" else null}"
    }
}

data class Feature(
    val featureName: String,
    val afterFeatures: List<String> = listOf(),
    val repository: Repository,
    val plugins: List<String> = listOf()
)
